contributions/api/controllers/contributions.py

```diff
@@ -0,0 +1,574 @@
+            "error": "Bad Request: " + request.url,
+        }
+        msg_json = jsonutils.create_log_json("Talent", "SEARCH", msg)
+        logging.error("Talent SEARCH " + json.dumps(msg_json))
+        return rs_handlers.bad_request(msg_json)
+
+    return_json = []
+    if out_json is None:
+        return_json = []
+    else:   # extract out talent with the given name
+        if isinstance(out_json, list):
+            for tmp_json in out_json:
+                talents_json = tmp_json["talents"]
+                # TODO this is the case of only 1 args that is name.
+                #  If there are more args this should be updated
+                for tmp_talent_json in talents_json:
+                    talent_json = None
+                    if tmp_talent_json["name"] == args["name"]:
+                        talent_json = tmp_talent_json
+                        return_json.append(talent_json)
+        else:
+            talents_json = out_json["talents"]
+            # TODO this is the case of only 1 args that is name.
+            #  If there are more args this should be updated
+            for tmp_talent_json in talents_json:
+                talent_json = None
+                if tmp_talent_json["name"] == args["name"]:
+                    talent_json = tmp_talent_json
+                    return_json.append(talent_json)
+    msg = {
+        "search": "Talent search performed with arguments of : " + str(args),
+        "result": return_json,
+    }
+    msg_json = jsonutils.create_log_json("Talent", "SEARCH", msg)
+    logging.info("Talent SEARCH " + json.dumps(msg_json))
+
+    return return_json
+
+def talents_search(id):
+    try:
+        contribution_dataset = mongoutils.get_contribution_dataset_from_objectid(coll_contribution, id)
+        talent_dataset = contribution_dataset.get_talents()
+    except Exception as ex:
+        msg = {
+            "reason": "There is no contribution dataset with given id: " + str(id),
+            "error": "Not Found: " + request.url,
+        }
+        msg_json = jsonutils.create_log_json("Talent", "SEARCH", msg)
+        logging.error("Talent SEARCH " + json.dumps(msg_json))
+        return rs_handlers.bad_request(msg_json)
+
+    if talent_dataset is None:
+        msg = {
+            "reason": "There is no Talent with given contribution id: " + str(id),
+            "error": "Not Found: " + request.url,
+        }
+        msg_json = jsonutils.create_log_json("Talent", "GET", msg)
+        logging.error("Talent SEARCH " + json.dumps(msg_json))
+        return rs_handlers.not_found(msg_json)
+
+    msg = {
+        "search": "Talent data in the contirubion dataset with given id : " + str(id),
+        "result": talent_dataset,
+    }
+    msg_json = jsonutils.create_log_json("Talent", "SEARCH", msg)
+    logging.info("Talent SEARCH " + json.dumps(msg))
+
+    return talent_dataset
+
+def construct_capability(in_json):
+    is_required_field = True
+    error_required = ""
+    try:
```

Comment:
If a required field is missing, this try block won't be executed. Connexion will return bad request error to client based on the API specification.

Reply:
Yes. I put this for like an insurance when the yaml was not really fixed, but I don't really remember 100 percent.